experimental/iterators/lib/Conversion/IteratorsToLLVM/IteratorsToLLVM.cpp

```diff
@@ -613,10 +613,10 @@ static Value buildCloseBody(FilterOp op, RewriterBase &rewriter,
 ///          !llvm.struct<"iterators.filter_state", (!nested_state)>
 static Value buildStateCreation(FilterOp op, RewriterBase &rewriter,
                                 LLVM::LLVMStructType stateType,
-                                ValueRange upstreamStates) {
+                                ValueRange operands) {
   Location loc = op.getLoc();
   Value undefState = rewriter.create<UndefOp>(loc, stateType);
-  Value upstreamState = upstreamStates[0];
+  Value upstreamState = operands[0];
```

Comment:
Can you use `FilterOpOperandAdapter(operands)` here and everywhere to avoid magic constants?
You may need to introduce the helper functions in the extraClass declaration in the op.
It was part of the LLVM cleanups / refactorings I wanted to do while you were away but never got to ..

Reply:
Done as part of #554.